models/comments.model.js

```diff
@@ -0,0 +1,28 @@
+const db = require("../db/connection");
+
+function selectCommentsByArticleId(article_id) {
+  return (
+    db
+      .query(
+        `
+    SELECT
+      comment_id,
+      votes,
+      created_at,
+      author,
+      body,
+      article_id
+    FROM comments
+    WHERE article_id = $1
+    ORDER BY created_at DESC
+    `,
+        [article_id]
+      )
+      // N.B. no need to test for empty rows as it is valid for no comments
+      // to be attached to an article. We can assume that article_id has been
+      // validated by controller before we query the database
+      .then((results) => results.rows)
+  );
```

Comment:
What happens if there are no comments? Ideally you would want a promise.reject here 💡

Reply:
Would a `Promise.reject` not result in an error rather than an empty array when the `article_id` is valid but there are no comments for it?